File: Python/Longest Substring Without Repeating Characters.py
"""
Given a string, find the length of the longest substring without repeating characters. For example, the longest substring without repeating letters for "abcabcbb" is "abc", which the length is 3. 
For "bbbbb" the longest substring is "b", with the length of 1.
"""

class Solution:
# @return an integer
    def lengthOfLongestSubstring(self, s):
        hash,i,Maxlen ={}, 0,0
        for j in range(len(s)):
            if hash.has_key(s[j]):
                Maxlen = max(Maxlen,j-i)
                while i < len(s) and s[j] != s[i]:
                    del hash[s[i]]
                    i +=1
                i+=1
            else:
                hash[s[j]] = j
        Maxlen = max(len(s)-i,Maxlen)
        return Maxlen

File: Java/Longest Substring Without Repeating Characters.java
/*
Given a string, find the length of the longest substring without repeating characters. For example, the longest substring without repeating letters for "abcabcbb" is "abc", which the length is 3. 
For "bbbbb" the longest substring is "b", with the length of 1.
*/

public class Solution {
    public int lengthOfLongestSubstring(String s) {
        int [] table = new int[256];
        Arrays.fill(table,-1);
        int start = 0;
        int result = 0;
        for(int j = 0;j<s.length();j++)
        {
            if(table[s.charAt(j)] >= start)
            {
                start = table[s.charAt(j)] + 1;
            }
            table[s.charAt(j)] = j;
            result = Math.max(result,j - start +1);
        }
        return result;
    }
}